airflow/models/mappedoperator.py

```diff
@@ -620,13 +620,19 @@ def expand_mapped_task(self, run_id: str, *, session: Session) -> tuple[Sequence
         try:
             total_length = self._get_specified_expand_input().get_total_map_length(run_id, session=session)
         except NotFullyPopulated as e:
-            self.log.info(
-                "Cannot expand %r for run %s; missing upstream values: %s",
-                self,
-                run_id,
-                sorted(e.missing),
-            )
-            total_length = None
+            if self.dag and self.dag.partial:
+                # partial dags comes from the mini scheduler. It's
+                # likely that the upstream tasks are not yet done,
+                # so we ignore this exception.
+                total_length = None
+            else:
+                self.log.info(
```

Comment:
What do you think @uranusjr.? I'm for changing it to error log

Reply:
This can be changed to WARNING if we are sure it’s only supposed to be called when the task is ready for expansion (i.e. all referenced upstreams have finished).

The DagRun implementation ensures this by ensuring expansion only happens when a task’s dependencies are all met: https://github.com/apache/airflow/blob/98a9c574e3e30b3aea710c23e18955e88871c9e6/airflow/models/dagrun.py#L744-L766

but I don’t quite recall if the backfill job does the same. Can someone track this down please?

Reply:
The dependency context ignores mapped task https://github.com/apache/airflow/blob/98a9c574e3e30b3aea710c23e18955e88871c9e6/airflow/models/dagrun.py#L738

Reply:
Yes but it still makes sure the task is _ready_ to be expanded (via `TriggerRuleDep`). I don’t think the mini scheduler does this; it simply tries to make downstreams until it succeeds.

Although looking at the implementation, `expand_mapped_task` will fail a task if expansion is attempted before it’s ready, so BackfillJob probably already does _something_ to make sure this does not happen (although I’m not sure what).

Let’s change this to WARNING (or maybe ERROR is better since it’s really not supposed to happen) and see what happens.